src/classes/showExpensesListForContactController.cls

```diff
@@ -0,0 +1,39 @@
+public class showExpensesListForContactController {
+    
+    private String queryString = '';
+    
+    public List<Expense__c> getExpenses(){
+        
+        
+        String idContact = ApexPages.currentPage().getParameters().get('id');
+            
+        List<Expense__c> results = Database.query('SELECT id, Name, Category__c, RecordType.Name, Amount__c, Tax__c, Client_Company__c, Submitted__c '+
```

Comment:
Same as before, please make sure you add a limit to your queries. You can even add pagination so that the user doesn't miss results in case your limit is overflown